setup.py

```diff
@@ -0,0 +1,19 @@
+from setuptools import setup, find_namespace_packages
+
+setup(
+    name="lsst.ts.salobj",
+    description="Object-oriented interface to LSST SAL communications middleware.",
+    use_scm_version=True,
+    setup_requires=["setuptools-scm", "pytest-runner"],
+    install_requires=[],
+    package_dir={"": "python"},
+    packages=find_namespace_packages(where="python"),
+    package_data={"": ["*.rst", "*.yaml"]},
+    scripts=["bin.src/purge_topics.py", "bin.src/run_test_csc.py", "bin.src/show_salpy_attributes.py"],
+    test_requires=["pytest", "pytest-cov", "pytest-flake8"],
+    license="GPL",
+    project_urls={
+        "Bug Tracker": "https://jira.lsstcorp.org/secure/Dashboard.jspa",
+        "Source Code": "https://github.com/lsst-ts/ts_salobj",
+    }
+)
```

Comment:
```
from setuptools import setup, find_namespace_packages
install_requires=["numpy"]
tests_require=["pytest", "pytest-cov", "pytest-flake8"]
setup(
    name="ts_salobj",
    description="Object-oriented interface to LSST SAL communications middleware.",
    use_scm_version={"write_to":"python/lsst/ts/salobj/version.py"},
    setup_requires=["setuptools-scm", "pytest-runner"],
    install_requires=install_requires,
    package_dir={"": "python"},
    packages=find_namespace_packages(where="python"),
    package_data={"": ["*.rst", "*.yaml"]},
    scripts=["bin.src/purge_topics.py", "bin.src/run_test_csc.py", "bin.src/show_salpy_attributes.py"],
    tests_require=tests_require,
    extras_require={"dev": install_requires + tests_require + ["documenteer[pipelines]"]},
    license="GPL",
    project_urls={
        "Bug Tracker": "https://jira.lsstcorp.org/secure/Dashboard.jspa",
        "Source Code": "https://github.com/lsst-ts/ts_salobj",
    }
)
```

This is what worked for me. I added an extras_require that allows the user to specify the dev dependencies. I also corrected the test_requires parameter to be tests_require parameter. Also numpy is an installation requirement. There is a version.py generated and so I have setuptools_scm write a version attribute to that module.

Reply:
OK. Pity about the confusing x_requires vs ys_require keywords but I see it's the standard.